.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: Release workflow
+
+on:
+  push:
+    branches: main
+
+env:
+  RUST_VERSION: rust:1.70.0
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    name: 'Detect changes'
+    outputs:
+      dockerfiles: ${{ steps.filter.outputs.changes }}
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            recover-id-object/scripts/Dockerfile:
+              - 'deps/concordium-rust-sdk/**'
+              - 'recover-id-object/**'
+
+  build-and-push:
+    needs: [ changes ]
+    if: needs.changes.outputs.dockerfiles != '[]'
+    strategy:
+      fail-fast: false
+      matrix:
+        dockerfile: ${{ fromJSON(needs.changes.outputs.dockerfiles) }}
+    runs-on: ubuntu-latest
+    environment: production
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+    - name: Log in to DockerHub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_TOKEN }}
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: ${{ matrix.dockerfile }}
+        push: true
+        build-args: |
+          build_image=${{ env.RUST_VERSION }}
```

Comment:
If this will not be the case in the future, lets handle it by then. So far it makes sense that we at least use the same rust version across different releases

Reply:
What do you mean? The contract here is that as long as we serve a single Dockerfile then everything we do is the same. Why do you want to replicate the same code for build release?